src/tests/tracing/eventlistener/EventListenerThreadPool.cs

```diff
@@ -71,24 +78,47 @@ static int Main()
+                // This should fire either of TPWorkerThreadStartCount, TPWorkerThreadStopCount or TPWorkerThreadWaitCount
+                int someNumber = 0;
+                Task[] tasks = new Task[100];
+                for (int i = 0; i < tasks.Length; i++) 
+                {
+                    tasks[i] = Task.Run(() => { someNumber += 1; });
+                }
 
-                // Test this part in CoreCLR only
-                if (!TestLibrary.Utilities.IsNativeAot)
+                if (TestLibrary.Utilities.IsWindows)
                 {
-                    int someNumber = 0;
-                    Task[] tasks = new Task[100];
-                    for (int i = 0; i < tasks.Length; i++)
+                    // This part is Windows-specific, it should fire an IOPack, IOEnqueue and IODequeue event
+                    listener.TPIOPackGoal += 1;
+                    listener.TPIOEnqueueGoal += 1;
+                    listener.TPIODequeueGoal += 1;
+                
+                    Overlapped overlapped = new Overlapped();
+                    unsafe
                     {
-                        tasks[i] = Task.Run(() => { someNumber += 1; });
+                        NativeOverlapped* nativeOverlapped = overlapped.Pack(null);
+                        ThreadPool.UnsafeQueueNativeOverlapped(nativeOverlapped);
                     }
+                }
+
+                // RegisterWaitForSingleObject should fire an IOEnqueue and IODequeue event
+                ManualResetEvent manualResetEvent = new ManualResetEvent(false);
+                WaitOrTimerCallback work = (x, timedOut) => { int y = (int)x; };
+                ThreadPool.RegisterWaitForSingleObject(manualResetEvent, work, 1, 100, true);
+                manualResetEvent.Set();
 
-                    listener.TPWaitWorkerThreadEvent.WaitOne(TimeSpan.FromMinutes(3));
+                ManualResetEvent[] waitEvents = new ManualResetEvent[] {listener.TPWaitIOPackEvent,
+                                                                        listener.TPWaitIOEnqueueEvent,
+                                                                        listener.TPWaitIODequeueEvent};
 
-                    bool workerThreadEventsOk = listener.TPWorkerThreadStartCount > 0 ||
-                                                listener.TPWorkerThreadStopCount > 0 ||
-                                                listener.TPWorkerThreadWaitCount > 0;
+                WaitHandle.WaitAll(waitEvents, TimeSpan.FromMinutes(1));
 
-                    if (!workerThreadEventsOk)
+                if (!TestLibrary.Utilities.IsNativeAot) // This test works in CoreCLR only, don't know why yet
+                {
+                    listener.TPWaitWorkerThreadEvent.WaitOne(TimeSpan.FromMinutes(1));
+                    if (listener.TPWorkerThreadStartCount == 0 &&
+                        listener.TPWorkerThreadStopCount == 0 &&
+                        listener.TPWorkerThreadWaitCount == 0)
```

Comment:
This test is likely not sufficient to guarantee raising the start/stop events, could remove checking for them